Terminal.Gui/Core/ConsoleDriver.cs

```diff
@@ -191,6 +277,44 @@ public static Attribute Get ()
 				throw new InvalidOperationException ("The Application has not been initialized");
 			return Application.Driver.GetAttribute ();
 		}
+
+		/// <summary>
+		/// Default empty attribute.
+		/// </summary>
+		public static readonly Attribute Default = new Attribute ();
+	}
+
+	/// <summary>
+	/// Defines a true color attribute.
+	/// </summary>
+	public class TrueColorAttribute : Attribute {
```

Comment:
I'm confused as to why this class is needed? Why not just put this into `Attribute` and make it so true color is the default/assumed, supporting non-truecolor as a fallback?

When would someone use this class?

Reply:
As describe before we need to decide which color escape sequence we should use. To keep backward compatibility i just introduced a new TrueColorAttribute class which holds the rgb values, like in the other PR.

This method has also he advantage that you only need to create new color escape sequences when the instance reference changes. However i can not evaluate if this has further side effects.

Furthermore there is no easy way to merge *basic* color and *truecolor* attribute class because any platform defines its own basic color range. The later must be unified first.

Reply:
Would it be possible to write a mapping algorithm for true color to Console Color?

An approximation would be fine, I'm guessing someone has already done this elsewhere.

If we could translate between the two then we could just add the properties to `Attribute` and add a new true color constructor.  So the pseudocode would be something like:

```csharp
public class Attribute
{
	/// <summary>
	/// Red color component.
	/// </summary>
	public int Red { get; }
	/// <summary>
	/// Green color component.
	/// </summary>
	public int Green { get; }
	/// <summary>
	/// Blue color component.
	/// </summary>
	public int Blue { get; }

	public Color Foreground { get; }
	public Color Background { get; }

	public Attribute (Color foreground, Color background)
	{
		Foreground = foreground;
		Background = background;
		var m = MapToTrueColor (foreground, background);
		Red = m [0];
		Green = m [1];
		Blue = m [2];
	}
	public Attribute (int red, int green, int blue)
	{
		Red = red;
		Green = green;
		Blue = blue;

		var b = MapToBasicColor (red,green,blue);			

		Foreground = b[0];
		Background = b[1];
	}

    [...]
}
```

That would allow clients to mix and match true color whenever they wanted.  And even have an app that supports true color but falls back on basic colors when deployed to an older machine (with zero effort on the API consumers behalf)?

Reply:
Yeah, something like that could work. Maybe we can track the index by the lifetime of the Attribute class and use a database approach to select a free index by incrementing a static _next-index_ value.

Reply:
Apparently [this is the RGB breakdown](https://stackoverflow.com/a/36674533/4824531).  I might pull your branch and see if a 'transparent' and backwards compatible approach can be adopted following this approach.

|Name    |     R |   G |   B|
|--------------|---|----|----|
|Black        |00 | 00  | 00   |
|DarkBlue     |00 | 00  | 80   |
|DarkGreen    |00 | 80  | 00   |
|DarkCyan     |00 | 80  | 80   |
|DarkRed      |80 | 00  | 00   |
|DarkMagenta  |80 | 00  | 80   |
|DarkYellow   |80 | 80  | 00   |
|DarkGray     |80 | 80  | 80   |
|Blue         |00 | 00  | FF   |
|Green        |00 | FF  | 00   |
|Cyan         |00 | FF  | FF   |
|Red          |FF | 00  | 00   |
|Magenta      |FF | 00  | FF   |
|Yellow       |FF | FF  | 00   |
|Gray         |C0 | C0  | C0   |
|White        |FF | FF  | FF   |

Reply:
@veeman I've started working on this approach in https://github.com/tznind/gui.cs/tree/true-color 

Once its ready (and assuming its viable) I will open a PR to your branch.